File: scheduling/scheduling-annotation-processor/src/test/java/ru/tinkoff/kora/scheduling/annotation/processor/controller/ScheduledJdkOnceTest.java
package ru.tinkoff.kora.scheduling.annotation.processor.controller;

import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleOnce;

import java.time.temporal.ChronoUnit;

public class ScheduledJdkOnceTest {
    @ScheduleOnce(delay = 100, config = "baseline", unit = ChronoUnit.SECONDS)
    public void baseline() {

    }

    @ScheduleOnce(delay = 100, unit = ChronoUnit.SECONDS)
    public void noConfig() {

    }

    @ScheduleOnce(config = "onlyConfig")
    public void onlyConfig() {

    }

    @ScheduleOnce(delay = 1000)
    public void onlyRequired() {

    }

    @ScheduleOnce(delay = 1000, config = "onlyRequiredWithConfig")
    public void onlyRequiredWithConfig() {

    }
}

File: scheduling/scheduling-ksp/src/test/kotlin/ru/tinkoff/kora/scheduling/ksp/controller/ScheduledJdkOnceTest.kt
package ru.tinkoff.kora.scheduling.ksp.controller

import ru.tinkoff.kora.scheduling.jdk.annotation.ScheduleOnce
import java.time.temporal.ChronoUnit

class ScheduledJdkOnceTest {
    @ScheduleOnce(delay = 100, config = "baseline", unit = ChronoUnit.SECONDS)
    fun baseline() {
    }

    @ScheduleOnce(delay = 100, unit = ChronoUnit.SECONDS)
    fun noConfig() {
    }

    @ScheduleOnce(config = "onlyConfig")
    fun onlyConfig() {
    }

    @ScheduleOnce(delay = 1000)
    fun onlyRequired() {
    }

    @ScheduleOnce(delay = 1000, config = "onlyRequiredWithConfig")
    fun onlyRequiredWithConfig() {
    }
}
